### PR TITLE
add service account and creds for pipelineRuns

### PR DESCRIPTION
**Pull request by jnunyez**
- PipelineRuns demand access to private repos or private registries.  The Service Account  associated to PipelineRuns must include credentials configured as Secrets. 
- The credentials embedded in the Secret objects are used to authenticate when needed private resources (e.g., a private repo or a private registry) demanded by the execution of a PipelineRun.
- Needed for sriov and ptp demos.

### CODE REVIEW

File: environments/overlays/cicd/pipeline-bot.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: build-bot
```
Comment (natifridman) [inline environments/overlays/cicd/pipeline-bot.yaml:5]: Maybe we can give this a more meaningful name. What about
```suggestion
  name: pipeline-credentials-sa
```

File: environments/overlays/cicd/pipeline-bot.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+secrets:
+  - name: github-cred
+  - name: quay-cred
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  creationTimestamp: null
+  name: cicd-admin
```
Comment (natifridman) [inline environments/overlays/cicd/pipeline-bot.yaml:14]: Same, this can be something like
```suggestion
  name: pipeline-credentials-rolebinding
```

**Review by natifridman — CHANGES_REQUESTED** [review]
Please fix lint errors.
Do we need here another applicationset?

**Comment by jnunyez** [comment]
> Please fix lint errors. Do we need here another applicationset?

Lint fixed. We totally do need that. This should be managed by ArgoCD. I was planning to add that one in another PR focused on Appset updates . 

I was thinking on Appset that provisions the cicd-creds and then the cicd catalog with all tasks and pipelines so that PipelineRuns consumes them.

File: environments/overlays/cicd/pipeline-bot.yaml (hunk 1 above)
Comment (jnunyez) [reply environments/overlays/cicd/pipeline-bot.yaml:5]: Fully agree. Taking note that all PipelineRuns must be changed in terms of Service Account to that new name.

File: environments/overlays/cicd/pipeline-bot.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (natifridman) [inline environments/overlays/cicd/pipeline-bot.yaml:1]: Maybe we can change the file name to something more meaningful so users can understand what it is.
WDYT about something like `pipeline-credentials.yaml`

**Review by natifridman — COMMENTED** [review]
Looks good, added one last comment

File: environments/overlays/cicd/pipeline-bot.yaml (hunk 3 above)
Comment (jnunyez) [reply environments/overlays/cicd/pipeline-bot.yaml]: Done. Needs to rebase before merge.

**Review by natifridman — APPROVED** [review]
LGTM, please squash the commits

**Pull request merged**